hashintel/hash
# Add button styles for HASH workspace

## 🌟 What is the purpose of this PR?



This PR adds the styling for the different button variants and states in our design system.

## ⚠️ Known issues




## 🐾 Next steps



## 🔍 What does this change?




- Removes `PageSidebar.module.css`, since PageSidebar styling is currently implemented with MUI
- Adds new button sizes to the theme and their respective styling
- Adds new button variants to the theme and styling (including hover/focus styles)
- Adds new SubmitButtonWrapper component with props `useTooltip` and `helperText` for showing helperText when the button is disabled.
- Add styles for loading state. Introduced 2 props => `loading` (shows loading spinner and "Loading..." text) and `loadingWithoutText` (just shows loading spinner)
- Merge `LinkButton` component into the Button component
- Added a `LoadingSpinnerIcon` component for loading states
- Added a new typography variant => `regularTextPages`
- Added a new box shadow variant => `purpleShadowMd`
- Replace deprecated `OldButton` with new `Button` component
- Adds new `IconButton` component which is similar to Mui's IconButton but has 2 new props => `unpadded` and `rounded`

### 📜 Does this require a change to the docs?



- No

## 🔗 Related links





- [Add styles for all button variants](https://app.asana.com/0/1201864704093982/1201916223398333) (_internal_)

## 🛡 What tests cover this?



- Manual and Existing tests

## ❓ How to test this?



1.  Checkout the branch 
1.  Login (after this you will be redirected to your account page)
1.  Confirm you see the different button variants similar to ones in attached video
1. Test out different props on the buttons in `layout/MainContentWrapper.tsx`

## 📹 Demo



https://user-images.githubusercontent.com/16928688/157628499-c2b630ff-7e18-4b3c-b4bf-d3a7c8358d3d.mov
